### ISSUE TITLE
Wrong matching number with custom rule's "search_in" property in "conditions"

### ISSUE THREAD

**Issue by jamongcandy** [microsoft/ApplicationInspector#422]
**Describe the bug**
Matching number of custom rule is higher than expected, when using "search_in" property in "conditions".

**To Reproduce**
`target.c` file
```c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <string.h>

#define BUFSIZER1 512
#define BUFSIZER2 ((BUFSIZER1 / 2) - 8)

int main(int argc, char **argv)
{
    char *buf1R1;
    char *buf2R1;
    buf1R1 = (char *)malloc(BUFSIZER1);
    buf2R1 = (char *)malloc(BUFSIZER1);
    free(buf2R1);
    strncpy(buf2R1, argv[1], BUFSIZER1 - 1);
    free(buf1R1);
}
```

`myRule.json` file
```json
[
    {
        "id": "SA000005",
        "name": "Testing.Rules.MallocAndFree",
        "tags": [
            "Testing.Rules.MallocAndFree"
        ],
        "severity": "Critical",
        "description": "this rule aims to find malloc() AND free() in 1 line range",
        "patterns": [
            {
                "pattern": "malloc\\(",
                "type": "regex",
                "confidence": "High",
                "modifiers": [
                    "i"
                ],
                "scopes": [
                    "code"
                ]
            }
        ],
        "conditions": [
            {
                "pattern": {
                    "pattern": "free\\(",
                    "type": "regex",
                    "scopes": [
                        "code"
                    ],
                    "modifiers": [
                        "i"
                    ]
                },
                "search_in": "finding-region(0,1)",
                "negate_finding": false
            }
        ],
        "_comment": ""
    }
]
```

terminal command
`appinspector analyze -i -s target.c -r myRule.json -f json -o output.json`

output
`output.json` file
```json
"detailedMatchList": [
      {
        "ruleId": "SA000005",
        "ruleName": "Testing.Rules.MallocAndFree",
        "ruleDescription": "this rule aims to find malloc() AND free() in 1 line range",
        "tags": [
          "Testing.Rules.MallocAndFree"
        ],
        "severity": "Critical",
        "pattern": "malloc\\(",
        "confidence": "High",
        "type": "Regex",
        "language": "c",
        "fileName": "target.c",
        "sample": "malloc(",
        "excerpt": "    char *buf1R1;\n    char *buf2R1;\n    buf1R1 = (char *)malloc(BUFSIZER1);\n    buf2R1 = (char *)malloc(BUFSIZER1);\n    free(buf2R1);\n    strncpy(buf2R1, argv[1], BUFSIZER1 - 1);\n    free(buf1R1);\n",
        "startLocationLine": 14,
        "startLocationColumn": 21,
        "endLocationLine": 14,
        "endLocationColumn": 28
      },
      {
        "ruleId": "SA000005",
        "ruleName": "Testing.Rules.MallocAndFree",
        "ruleDescription": "this rule aims to find malloc() AND free() in 1 line range",
        "tags": [
          "Testing.Rules.MallocAndFree"
        ],
        "severity": "Critical",
        "pattern": "malloc\\(",
        "confidence": "High",
        "type": "Regex",
        "language": "c",
        "fileName": "target.c",
        "sample": "malloc(",
        "excerpt": "{\n    char *buf1R1;\n    char *buf2R1;\n    buf1R1 = (char *)malloc(BUFSIZER1);\n    buf2R1 = (char *)malloc(BUFSIZER1);\n    free(buf2R1);\n    strncpy(buf2R1, argv[1], BUFSIZER1 - 1);\n",
        "startLocationLine": 13,
        "startLocationColumn": 21,
        "endLocationLine": 13,
        "endLocationColumn": 28
      }
    ],
```

My intention was to find `malloc()` AND `free()` that are in 1 line range.
The expected result was 1 pattern match, which is `malloc()` at line 14.

Instead, it gave result of 2 matches, `malloc()` at line 14 and line 13.
However, `malloc()` at line 13 should not be a match, since there is no `free()` in next line.(`"search_in": "finding-region(0,1)"`).

**Expected behavior**
1 Match found.

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04
 - ApplicationInpector Version: ApplicationInspector.CLI 1.4.22+ccdc2c44b5

**Comment by gfs** [comment]
Thanks for the well written report. I'll get a fix out soon.

**Comment by gfs** [comment]
This should be fixed in 1.4.23 (building in pipeline now, should release within an hour). Thanks again for your reports.
